app/main.py

```diff
@@ -30,3 +42,32 @@ def profile(request: Request):
         "username": current_username,
         "events": upcouming_events
     })
+
+
+@app.get("/agenda")
+def agenda(
+    request: Request,
+    start_date: Optional[date]=None,
+    end_date: Optional[date]=None,
+    days: Optional[int]=None) -> Jinja2Templates:
+    """Route for the agenda page, using dates range or exact amount of days."""
+    
+    user_id = 1   # there is no user session yet, so I use user id- 1.
+    start_date, end_date = agenda_events.calc_dates_range_for_agenda(start_date, end_date, days)
+
+    db_session = SessionLocal()
```

Comment:
Read https://fastapi.tiangolo.com/tutorial/sql-databases/ to learn about handling db in fastapi